types/coin_test.go

```diff
@@ -1276,6 +1276,40 @@ func (s *coinTestSuite) TestMarshalJSONCoins() {
 	}
 }
 
+func (s *coinTestSuite) TestCoinValidate() {
+	testCases := []struct {
+		name    string
+		coin    sdk.Coin
+		wantErr string
+	}{
+		{"nil coin: nil Amount", sdk.Coin{}, "invalid denom"},
+		{"non-blank coin, nil Amount", sdk.Coin{Denom: "atom"}, "amount is nil"},
+		{"valid coin", sdk.Coin{Denom: "atom", Amount: math.NewInt(100)}, ""},
+		{"valid coin negative", sdk.Coin{Denom: "atom", Amount: math.NewInt(-999)}, "negative coin amount"},
```

Comment:
nit: I wouldn't say it is a valid coin here.

Reply:
Thanks for pointing it out @julienrbrt, addressed!